plugins/outputs/azure_loganalytics/azure_loganalytics.go

```diff
@@ -0,0 +1,237 @@
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+var sampleConfig = `
+  ## Customer ID (Workstation ID) and Key for Azure Log Analytics resource.
+  # customer_id = "<Workstation ID>"
+  # shared_key = "<Secret>"
+
+  ## Timeout for closing (default: 5s).
+  # timeout = "5s"
+
+  ## Table Namespace Prefix (default: "").
+  ## Namespace Prefix is used in "Log-Type" header
+  ## Prefix can only contain alphaNumeric characters
+  # namespace_prefix = ""
+`
+
+const (
+	baseURL                = "https://%s.ods.opinsights.azure.com/api/logs?api-version=2016-04-01"
+	contentType            = "application/json"
+	timeGeneratedFieldName = "DateTime"
+	namespacePrefixRegex   = "^[a-zA-Z0-9]*$"
+	httpMethod             = http.MethodPost
+
+	defaultClientTimeout = 5 * time.Second
+)
+
+// AzLogAnalytics contains information about a azure log analytics service metadata
+type AzLogAnalytics struct {
+	CustomerID string `toml:"customer_id"`
+	SharedKey  string `toml:"shared_key"`
+
+	NamespacePrefix string            `toml:"namespace_prefix"`
+	ClientTimeout   internal.Duration `toml:"timeout"`
+
+	serviceURL string
+	client     *http.Client
+}
+
+// Connect initializes the plugin and validates connectivity
+func (a *AzLogAnalytics) Connect() error {
+
+	if a.CustomerID == "" {
+		return fmt.Errorf("customer_id not configured")
+	}
+
+	if a.SharedKey == "" {
+		return fmt.Errorf("shared_key not configured")
+	}
+
+	if len(a.NamespacePrefix) > 25 {
+		return fmt.Errorf("namespace_prefix length is greater than 25 characters")
+	}
+
+	if a.ClientTimeout.Duration == 0 {
+		a.ClientTimeout.Duration = defaultClientTimeout
+	}
+
+	match, err := regexp.MatchString(namespacePrefixRegex, a.NamespacePrefix)
+	if err != nil {
+		return err
+	} else if !match {
+		return fmt.Errorf("namespace_prefix contains invalid characters")
+	}
+
+	ctx := context.Background()
+	client, err := a.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	a.client = client
+	a.serviceURL = fmt.Sprintf(baseURL, a.CustomerID)
+
+	return nil
+}
+
+// Close shuts down an any active connections
+func (a *AzLogAnalytics) Close() error {
+	return nil
+}
+
+// Description provides a description of the plugin
+func (a *AzLogAnalytics) Description() string {
+	return "A plugin that can transmit metrics to Azure Log Analytics"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (a *AzLogAnalytics) SampleConfig() string {
+	return sampleConfig
+}
+
+// Write writes metrics to the remote endpoint
+func (a *AzLogAnalytics) Write(metrics []telegraf.Metric) error {
+
+	objects := make(map[string][]interface{}, len(metrics))
+	for _, metric := range metrics {
+		m, name := a.createObject(metric)
+		objects[name] = append(objects[name], m)
+	}
+
+	for name, data := range objects {
+		serialized, err := json.Marshal(data)
+		if err != nil {
+			return err
+		}
+
+		if err := a.write(name, serialized); err != nil {
```

Comment:
Just an observations, this will be one request per metric name, which seems like it will be rather expensive.  I'm not sure what we can do about this, since we need to have different Log-Type to avoid the 500 field limitations.

Reply:
We should not that and try to improve this with batching.